export_all.py

```diff
@@ -0,0 +1,218 @@
+# MIT License
+#
+# Copyright (c) 2018 Evgeny Medvedev, evge.medvedev@gmail.com
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import argparse
+import csv
+import os
+
+from time import time
+from datetime import datetime
+from web3 import Web3
+
+from ethereumetl.jobs.export_blocks_job import ExportBlocksJob
+from ethereumetl.jobs.export_token_transfers_job import ExportTokenTransfersJob
+from ethereumetl.jobs.export_receipts_job import ExportReceiptsJob
+from ethereumetl.jobs.export_contracts_job import ExportContractsJob
+from ethereumetl.jobs.export_tokens_job import ExportTokensJob
+
+from ethereumetl.jobs.exporters.blocks_and_transactions_item_exporter import blocks_and_transactions_item_exporter
+from ethereumetl.jobs.exporters.token_transfers_item_exporter import token_transfers_item_exporter
+from ethereumetl.jobs.exporters.receipts_and_logs_item_exporter import receipts_and_logs_item_exporter
+from ethereumetl.jobs.exporters.contracts_item_exporter import contracts_item_exporter
+from ethereumetl.jobs.exporters.tokens_item_exporter import tokens_item_exporter
+
+from ethereumetl.logging_utils import logging_basic_config
+from ethereumetl.providers.auto import get_provider_from_uri
+from ethereumetl.thread_local_proxy import ThreadLocalProxy
+from ethereumetl.csv_utils import set_max_field_size_limit
+from ethereumetl.file_utils import smart_open
+
+parser = argparse.ArgumentParser(description='Export all for a range of blocks.',
+                                 usage='-s <start_block> -e <end_block> -b <batch_size> -p <provider_uri> [-o <output_dir>]')
```

Comment:
Since the `-o` argument is required, the square brackets around `-o <output_dir>` should be removed. Alternatively you can specify default value for `-o`

Reply:
- I'll specify `default=None` (to match with the original shell scripts).
- Another point I forgot to mention is that I pulled out `--max-workers` in my code (which was lacking in the original) so I'll add that as well -> `'-s <start_block> -e <end_block> -b <batch_size> -p <provider_uri> [-o <output_dir> -w <max_workers>]'`